src/ReactLeafletMap/index.js

```diff
@@ -275,10 +272,56 @@ class ReactLeafletMap extends React.PureComponent {
         );
     }
 
-    getIndexedVectorLayer(layer, i, isDiagram) {
+    getVectorLayer(layer, i, zIndex) {
+    	const renderAs = layer.options?.renderAs;
+    	if (renderAs) {
+    		const boxRange  = this.state.view?.boxRange || this.props.view?.boxRange;
+    		const renderAsData = _.find(renderAs, (renderAsItem) => {
+    			const boxRangeRange = renderAsItem.boxRangeRange;
+
+    			// Current boxRange is in defined range
+    			return (boxRange > boxRangeRange[0] && boxRange <= boxRangeRange[1]) || (!boxRangeRange[0] && boxRange <= boxRangeRange[1]) || (boxRange > boxRangeRange[0] && !boxRangeRange[1]);
+			});
+
+    		if (renderAsData && renderAsData.type) {
+    			// TODO enable to define other layer options in renderAs
+				const options = {
+					...layer.options,
+					style: renderAsData.options?.style || layer.options?.style,
+					pointAsMarker: renderAsData.options?.hasOwnProperty("pointAsMarker") ? renderAsData.options.pointAsMarker : layer.options?.pointAsMarker,
+				}
+
+				return this.getVectorLayerByGivenType(renderAsData.type, layer, i, zIndex, options);
+			} else {
+				return this.getVectorLayerByGivenType(layer.type, layer, i, zIndex, layer.options);
+			}
+		} else {
+			return this.getVectorLayerByGivenType(layer.type, layer, i, zIndex, layer.options);
```

Comment:
It is same like on line 286

Reply:
You mean that line 296 is the same as 299? It is. However, for me is this more readable

Reply:
Yes, there are two else with the same content. It looks... wrong. BUT ok

Reply:
Could you propose better solution?

Reply:
Actualy not :) True is, that every "if" should has "else" block.